### PR TITLE
Fixes various issues with the choropleth map

### PR DESCRIPTION
**Pull request by IanCostello**
1. Prevents hard failures in many API calls with better handling or failure conditions.
2. Fixes the on-hover code so that subgeographies are again highlighted (feature was broken)
3. For region with subgeos, a hand pointer is displayed to indicate that you can click into the subgeo. 
4. The map no longer displays an error if no statistical variable is chosen and a message is displayed.
5. API call failures now alert the user to the failure instead of displaying loading forever

### CODE REVIEW

File: static/js/tools/choropleth/choropleth.tsx [hunk 1]
```diff
@@ -72,7 +70,9 @@ class ChoroplethMap extends Component {
       //shouldComponentUpdate.
       this.renderGeoMap();
       this.updateGeoValues();
-    });
+    }, () => {
+      document.getElementById("heading").innerHTML = "API Request Failed!"
```
Comment (beets) [inline static/js/tools/choropleth/choropleth.tsx:74]: whoa! that's a loud error... could you move these error messages to a different div that we can easily style later?

Comment (beets) [reply static/js/tools/choropleth/choropleth.tsx:74]: also, error messages are generally more helpful when there's a recourse for the user. what should they do to get things working again?

Comment (beets) [reply static/js/tools/choropleth/choropleth.tsx:74]: applies to the rest of the error messages as well

Comment (IanCostello) [reply static/js/tools/choropleth/choropleth.tsx:74]: I added better error messages for both of the UI and moved these over to a new div.

File: server/routes/api/choropleth.py [hunk 2]
```diff
@@ -59,7 +59,8 @@ def choropleth_values():
                                  flask.request.args.get("level"))
     if not display_level:
         return flask.jsonify({"error":
-           f"Failed to automatically resolve subgeo for {dcid}."}, 400) 
+           f"Failed to automatically resolve subgeo for {dcid}." +
```
Comment (jeffreyoldham) [inline server/routes/api/choropleth.py:62]: "subgeo" might not be an easily understood term.  I am not sure what a good term is.  ""geographic subdivision level" is a guess.

**Pull request merged**